File: Binary Tree Inorder Traversal/Binary tree inorder traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    List<Integer> answer = new ArrayList<>();
    
    public List<Integer> inorderTraversal(TreeNode root) {
        inorder(root);
        return this.answer;
    }
    
    public void inorder(TreeNode node) {
        if (node != null) {
            inorder(node.left);
            this.answer.add(node.val);
            inorder(node.right);
        }
    }
}

File: Binary Tree Inorder Traversal/Binary tree inorder traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> answer; 
        inorder(root, answer);
        return answer;
    }
    
    void inorder(TreeNode* node, vector<int>& answer) {
        if (node != NULL) {
            inorder(node->left, answer);
            answer.push_back(node->val);
            inorder(node->right, answer);
        }
    }
};